pkg/ebpf/offsetguess.go

```diff
@@ -106,46 +122,64 @@ type fieldValues struct {
 	rtt       uint32
 	rttVar    uint32
 	daddrIPv6 [4]uint32
+
+	// Used for guessing offsets in struct flowi4
+	saddrFl4 uint32
+	daddrFl4 uint32
+	sportFl4 uint16
+	dportFl4 uint16
 }
 
-func expectedValues(conn net.Conn) (*fieldValues, error) {
-	netns, err := ownNetNS()
+func extractIPsAndPorts(conn net.Conn) (
+	saddr, daddr uint32,
+	sport, dport uint16,
+	err error,
+) {
+	saddr_s, sport_s, err := net.SplitHostPort(conn.LocalAddr().String())
```

Comment:
nit: underscores are not something we usually use for variable names in Go. Camelcasing is preferred, e.g. `saddrStr`